### PR TITLE
fix: include attachment in the runfiles

### PR DESCRIPTION
**Pull request by thesayyn**

### CODE REVIEW

File: cosign/private/attach.bzl [hunk 1]
```diff
@@ -54,11 +54,11 @@ def _cosign_attach_impl(ctx):
     fixed_args = ["--repository", ctx.attr.repository]
 
     if ctx.attr.type == "sbom":
-        fixed_args.extend(["--sbom", ctx.file.attachment.path])
+        fixed_args.extend(["--sbom", ctx.file.attachment.short_path])
```
Comment (alexeagle) [inline cosign/private/attach.bzl:57]: Why this change?

File: examples/attach/test.bash [hunk 2]
```diff
@@ -1,6 +1,8 @@
 #!/usr/bin/env bash
 set -o pipefail -o errexit -o nounset
 
+export HOME="$TEST_TMPDIR"
```
Comment (alexeagle) [inline examples/attach/test.bash:4]: Should there be an added assertion?

File: cosign/private/attach.bzl (hunk 1 above)
Comment (thesayyn) [reply cosign/private/attach.bzl:57]: these go into runfiles where it needs to refer to the file by its short_path because the path contains output tree segments.  this never worked for generated files but worked for source files since those have identical short_path and path.

Comment (thesayyn) [reply cosign/private/attach.bzl:57]: https://github.com/bazel-contrib/rules_oci/blob/982972a830655537dd1ff8f85f6815c52410e50b/cosign/private/sign.bzl#L60 uses short_path as both should

File: examples/attach/test.bash (hunk 2 above)
Comment (thesayyn) [reply examples/attach/test.bash:4]: I don't think we need one for this. it is there because cosign wants to read credentials which unrestricted at `bazel run`

File: cosign/private/attach.bzl (hunk 1 above)
Comment (alexeagle) [reply cosign/private/attach.bzl:57]: if it's for a runfiles lookup, then I think you need our `to_manifest_path` helper, short_path isn't always correct. e.g. https://github.com/aspect-build/rules_esbuild/pull/108 or https://github.com/aspect-build/rules_js/blob/0b4e900ea299ea6e77d45f1f2b5f03167bb7f3c0/js/private/js_image_layer.bzl#L134

Comment (thesayyn) [reply cosign/private/attach.bzl:57]: I am not sure if we need this. I have added a test case where we sign an oci_image target from an external workspace.

File: examples/attach/BUILD.bazel [hunk 3]
```diff
@@ -10,7 +17,7 @@ oci_image(
 
 cosign_attach(
     name = "attach",
-    attachment = "sbom.spdx",
+    attachment = ":sbom_generated.spdx",
     image = ":image",
     repository = "test",
     type = "sbom",
```
Comment (alexeagle) [inline examples/attach/BUILD.bazel:23]: don't you need to use the external repo in one of these attributes to see a potential failure?

Comment (thesayyn) [reply examples/attach/BUILD.bazel:23]: I added the test for signing. I'll add a test for this one as well.

Comment (thesayyn) [reply examples/attach/BUILD.bazel:23]: done

**Review by alexeagle — APPROVED** [review]
